src/index.js

```diff
@@ -32,19 +32,21 @@ const plugin = {
 		"no-empty-keys": noEmptyKeys,
 		"no-unsafe-values": noUnsafeValues,
 	},
-	configs: {},
-};
-
-Object.assign(plugin.configs, {
-	recommended: {
-		plugins: { json: plugin },
-		rules: {
-			"json/no-duplicate-keys": "error",
-			"json/no-empty-keys": "error",
-			"json/no-unsafe-values": "error",
+	configs: {
+		recommended: {
+			plugins: {
+				get json() {
```

Comment:
> Maybe we could leave `plugins` as an empty object here and just assign `json` into the object later?

With that change, TypeScript code will not be able to access `plugin.configs.recommended.plugins.json`. I think that's fine since consumers should not have to look into that property but I'm not totally sure.

Another option to expose `plugin.configs.recommended.plugins.json` with the correct type is using a getter in the `plugin` declaration and replacing it later with a plain value using `Reflect.defineProperty`.

Finally we could override the type inferred by tsc with a custom type (or interface) for `plugin`. That would make maintenance more complicated though because we'd need to update the types every time something in the plugin changes, so probably not our favorite solution.

Reply:
> With that change, TypeScript code will not be able to access `plugin.configs.recommended.plugins.json`. I think that's fine since consumers should not have to look into that property but I'm not totally sure.

Yeah, I don't think we should overthink this. The most important thing is that consumers can see `configs.recommended`. There's no good reason to reach into the config object to look at what's actually contained in the `plugins` key.

Reply:
Thanks! Updated in 21f751ab535ba28770cefe30e89b403a6632c536.